clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -498,6 +498,7 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
 
         if (!outOfMemory && memoryPool.availableMemory() < lowMemThreshold) {
             List<SelectionKey> temp = new ArrayList<>(selectionKeys);
```

Comment:
Doesn't look like we need this `temp` variable. We can just assign it to `inHandlingOrder` directly, right? Also, using an `ArrayList` seems suboptimal given that we call `remove` on each element during iteration (each time shifting all the elements). `ArrayDeque` would work well if we didn't remove via the `Iterator`. Maybe we should change `pollSelectionKeys` to clear the collection at the end instead?

Reply:
I think the `temp` was possibly added since `Collections.shuffle()` needs a list. I have left that as-is. Since we only need to clear the selected key set, I have added `clear` to the end of `poll()` and got rid of `Iterator.remove`. It makes the code clearer as well.